src/models/stats_collection.rs

```diff
@@ -0,0 +1,67 @@
+use context::GraphQLContext;
+use juniper::{FieldResult, ID};
+use models::*;
+
+pub struct StatsCollection {
+    pub song_id: ID,
+    pub album_id: Option<ID>,
+    pub artist_id: Option<ID>,
+    pub playlist_id: Option<ID>,
+}
+
+impl StatsCollection {
+    pub fn song(&self, context: &GraphQLContext) -> FieldResult<Song> {
+        Song::from_id(context, self.song_id.as_ref())
```

Comment:
`self.song_id.as_ref()` could be `&self.song_id`

Reply:
Why does this work? `&self.song_id` is a `&ID`

Reply:
`ID` dereferences into `&str`

Reply:
Make these changes if you want. I think my way makes more sense.